core/src/com/unciv/logic/automation/civilization/NextTurnAutomation.kt

```diff
@@ -321,6 +321,98 @@ object NextTurnAutomation {
+                                (civInfo.getCivResourcesByName()[it.resource!!] ?: 0) <= 3)
+                it.isVisible(civInfo) && it.getOwner() == null &&
+                        (hasNaturalWonder || hasLuxuryCivDoesntOwn || hasResourceCivHasNoneOrLittle)
+            }
+            for (highlyDesirableTileInCity in highlyDesirableTilesInCity) {
+                highlyDesirableTiles.getOrPut(highlyDesirableTileInCity) { mutableSetOf() }
+                    .add(city)
+            }
+        }
+
+        // Always try to buy highly desirable tiles if it can be afforded.
+        for (highlyDesirableTile in highlyDesirableTiles) {
+            val cityWithLeastCostToBuy = highlyDesirableTile.value.minBy {
+                it.getCenterTile().aerialDistanceTo(highlyDesirableTile.key)
+            }
+            val bfs = BFS(cityWithLeastCostToBuy.getCenterTile()) {
+                it.getOwner() == null || it.getOwner() == civInfo
+            }
+            bfs.stepUntilDestination(highlyDesirableTile.key)
+            val tilesThatNeedBuying =
+                    bfs.getPathTo(highlyDesirableTile.key).filter { it.getOwner() != civInfo }
+
+            // We're trying to acquire everything and revert if it fails, because of the difficult
+            // way how tile acquisition cost is calculated. Everytime you buy a tile, the next one
+            // gets more expensive and by how much depends on other things such as game speed. To
+            // not introduce hidden dependencies on that and duplicate that logic here to calculate
+            // the price of the whole path, this is probably simpler.
+            var ranOutOfMoney = false
+            var goldSpent = 0
+            for (tileThatNeedsBuying in tilesThatNeedBuying) {
+                val goldCostOfTile =
+                        cityWithLeastCostToBuy.expansion.getGoldCostOfTile(tileThatNeedsBuying)
+                if (civInfo.gold >= goldCostOfTile) {
+                    cityWithLeastCostToBuy.expansion.buyTile(tileThatNeedsBuying)
+                    goldSpent += goldCostOfTile
+                } else {
+                    ranOutOfMoney = true
+                }
+            }
+            if (ranOutOfMoney) {
+                for (tileThatNeedsBuying in tilesThatNeedBuying) {
+                    cityWithLeastCostToBuy.expansion.relinquishOwnership(tileThatNeedsBuying)
+                }
+                civInfo.addGold(goldSpent)
+            }
+        }
+
+        // After highly desirable tiles, just buy the next tile the city expansion would go to if
+        // it doesn't consume too much of the total wealth. One could make this a lot more complex,
+        // but this is also taking cycles on the automation, so it might not be worth it. Things I
+        // considered where whether enemy cities / settlers are close etc., but I think this is a
+        // good approximation.
+        if (civInfo.gold > 500) {
```

Comment:
I don't agree with this at all. That'a a lot of gold spent for tiles that are potentially useless.

I'd rather the AI keep the gold for buting buildings/units, unless we filter this down to "find least productive citizen - find most productive tile - buy if potential tile production is better than current citizen production"  
But that may be something for another PR.